File: Wahtu/LeetCode/645.cpp
class Solution 
{
public:
    vector<int> findErrorNums(vector<int>& nums) 
    {
        int s = nums.size()*(nums.size()+1) / 2;
        
        int a = 0, u = 0;
        vector<int> fl(nums.size()+1, 1);
        for (auto n : nums)
        {
            a += n; 
            u += n*(fl[n]--);
        }
        
        return {a-u, s-u};
    }
};

File: Wahtu/LeetCode/645.go
func findErrorNums(nums []int) []int {
    mp := make(map[int]int)
    ans := make([]int, 0)
    n := len(nums)

    for i := 1; i <= n;i++ {
        mp[i] = 0
    }

    for _, num := range nums {
        if mp[num] == 1 {
            ans = append(ans, num)
        }
        mp[num]++
    }

    for i := 1; i <= n;i++ {
        if mp[i] == 0 {
            ans = append(ans, i)
        }
    }

    return ans
}
